scripts/test-runner-filter.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+try {
+  const minimatch = require('minimatch');
+  const childProcess = require('child_process');
+  const gitDiffRange = (
+    process.env.TRAVIS_COMMIT_RANGE || 'master..HEAD'
+  ).replace('...', '..');
+  const changedFiles = childProcess
+    .execSync(`git diff --name-only ${gitDiffRange}`, {encoding: 'utf8'})
+    .split('\n')
+    .slice(0, -1);
+
+  const testRelatedFilePatterns = [
+    'scripts/test-runner-filter.js',
+    'bin/**/*',
+    'lib/**/*',
+    'test/**/*',
+    '.mocharc.yml',
+    '.wallaby.js',
+    'index.js',
+    'package.json',
+    'package-lock.json',
+    'package-scripts.json',
+    '.nycrc',
+    '.travis.yml',
+    'browser-entry.js',
+    'karma.conf.js',
+    'mocha.*'
+  ];
+
+  const shouldRunTest = changedFiles.some(filePath =>
+    testRelatedFilePatterns.some(pattern => minimatch(filePath, pattern))
+  );
+
+  if (!shouldRunTest) {
+    console.log('No library related files have changed. Skipping test');
+    process.exit(1);
+  }
+
+  console.log('Library related files have changed. Continuing full tests');
+} catch (err) {
+  console.log(
+    'Caught an error while checking git diff against test file patterns:'
+  );
+  console.log(err);
+  console.log('Running the full test suite as a consequence');
+}
```

Comment:
Wouldn't this file be better named "scripts/travis-run-filter.js" (or something that includes "travis" in the name)?

Reply:
agree